### PR TITLE
pfSense support with ssh exec

### PR DESCRIPTION
**Pull request by stokbaek**

### CODE REVIEW

**Comment by ytti** [comment]
Without the ```prompt``` telnet won't work. It does not affect the ssh at all having it there.

So either
  1. remove telnet support (cfg :telnet)
  2. add prompt

File: lib/oxidized/pfsense.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class PfSense < Oxidized::Model
+  
+  comment  '# '
+  
+  #add a comment in the final conf
+  def add_comment comment
```
Comment (danilopopeye) [inline lib/oxidized/pfsense.rb]: why not use the `#comment` method directly?

File: lib/oxidized/pfsense.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    "\n###### #{comment} ######\n" 
+  end
+
+  cmd :all do |cfg|
+    cfg.each_line.to_a[1..-2].join
+  end
+  
+  #show the persistent configuration
+  pre do
```
Comment (danilopopeye) [inline lib/oxidized/pfsense.rb]: @stokbaek care to explain why this model is so different?
couldn't we just follow the others with something like:

``` ruby
cmd 'cat /cf/conf/config.xml'
```

**Comment by stokbaek** [comment]
I used the this cumulus (https://github.com/ytti/oxidized/blob/master/lib/oxidized/model/cumulus.rb) as a base when I made it as this is my first attempt to create a profile for oxidized. If it can be done in a more simple way then I see no issue in changing it.

**Comment by OBenned** [comment]
When i try and use this the only thing that happens is oxidized hangs with: 

D, [2016-08-19T16:26:31.543146 #2433] DEBUG -- : lib/oxidized/worker.rb: Added xxx.xxx.xxx.xxx to the job queue
D, [2016-08-19T16:26:31.543188 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:31.543294 #2433] DEBUG -- : lib/oxidized/job.rb: Starting fetching process for xxx.xxx.xxx.xxx at 2016-08-19 14:26:31 UTC
D, [2016-08-19T16:26:31.543486 #2433] DEBUG -- : lib/oxidized/input/ssh.rb: Connecting to xxx.xxx.xxx.xxx
D, [2016-08-19T16:26:31.712795 #2433] DEBUG -- : lib/oxidized/input/cli.rb: Running post_login commands at xxx.xxx.xxx.xxx
D, [2016-08-19T16:26:31.712901 #2433] DEBUG -- : lib/oxidized/model/model.rb Collecting commands' outputs
D, [2016-08-19T16:26:31.712976 #2433] DEBUG -- : lib/oxidized/model/model.rb Executing cat /cf/conf/config.xml
D, [2016-08-19T16:26:31.713024 #2433] DEBUG -- : lib/oxidized/input/ssh.rb cat /cf/conf/config.xml @ xxx.xxx.xxx.xxx with expect: /^([\w.@-]+[#>]\s?)$/
D, [2016-08-19T16:26:32.543592 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:33.543922 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:34.544262 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:35.544553 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:36.544841 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:37.545133 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:38.545448 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:39.545823 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:40.546136 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:41.546498 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:42.546888 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:43.547298 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:44.547661 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:45.548076 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:46.548422 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:47.548747 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:48.549081 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:49.549388 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:50.549781 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:51.550123 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:52.550443 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:53.550780 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:54.551127 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:55.551478 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:56.551846 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:57.552191 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:58.552524 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:26:59.552850 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-08-19T16:27:00.553199 #2433] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel

**Comment by ytti** [comment]
ping @stokbaek

**Comment by stokbaek** [comment]
@OBenned I have tried to reproduce your error and I cant seem to repreduce it. This is the output I get when I run it. I have tried both with custom user and with root:

```
D, [2016-08-19T17:05:05.300556 #15341] DEBUG -- : lib/oxidized/job.rb: Starting fetching process for X.X.X.X at 2016-08-19 15:05:05 UTC
D, [2016-08-19T17:05:05.300753 #15341] DEBUG -- : lib/oxidized/input/ssh.rb: Connecting to X.X.X.X
D, [2016-08-19T17:05:05.426531 #15341] DEBUG -- : lib/oxidized/input/cli.rb: Running post_login commands at X.X.X.X
D, [2016-08-19T17:05:05.426620 #15341] DEBUG -- : lib/oxidized/model/model.rb Collecting commands' outputs
D, [2016-08-19T17:05:05.426688 #15341] DEBUG -- : lib/oxidized/model/model.rb Executing cat /cf/conf/config.xml
D, [2016-08-19T17:05:05.426740 #15341] DEBUG -- : lib/oxidized/input/ssh.rb cat /cf/conf/config.xml @ X.X.X.X with expect: /^([\w.@-]+[#>]\s?)$/
D, [2016-08-19T17:05:05.476207 #15341] DEBUG -- : lib/oxidized/input/cli.rb Running pre_logout commands at X.X.X.X
D, [2016-08-19T17:05:05.476514 #15341] DEBUG -- : lib/oxidized/node.rb: Oxidized::SSH ran for X.X.X.X successfully
D, [2016-08-19T17:05:05.476593 #15341] DEBUG -- : lib/oxidized/job.rb: Config fetched for X.X.X.X at 2016-08-19 15:05:05 UTC
D, [2016-08-19T17:05:06.302788 #15341] DEBUG -- : lib/oxidized/worker.rb: Jobs 0, Want: 1
D, [2016-08-19T17:05:07.303207 #15341] DEBUG -- : lib/oxidized/worker.rb: Jobs 0, Want: 1
D, [2016-08-19T17:05:08.303612 #15341] DEBUG -- : lib/oxidized/worker.rb: Jobs 0, Want: 1
D, [2016-08-19T17:05:09.303933 #15341] DEBUG -- : lib/oxidized/worker.rb: Jobs 0, Want: 1
```
Looks like the ```D, [2016-08-19T17:05:06.302788 #15341] DEBUG -- : lib/oxidized/worker.rb: Jobs 0, Want: 1``` is something oxidized is ment to do. Correct me if I am wrong though @ytti .

**Comment by OBenned** [comment]
@stokbaek 
Yeah im pretty sure that DEBUG -- : lib/oxidized/worker.rb: Jobs 0, Want: 1 is something oxidized is doing while its idle, waiting for the next interval.
But thats not the samme im getting, mine says: DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel

Can it be because when you connect to the pfsense, its initially in the interactive menu or am i completly off?

**Comment by danilopopeye** [comment]
@OBenned can you turn on the input debug and paste here the log:

``` yaml
input:
  debug: true
```

then a file should be create at `[oxidized-dir]/logs/[device-ip]-ssh`

**Comment by OBenned** [comment]
I can for sure create a issue, but i enabled the input debug log and it is completely empty, i must be doing something totally wrong.

**Comment by OBenned** [comment]
I can for sure create a issue, but i have turned on the input debug log and the log is completely empty, i must be doing something wrong

**Pull request merged**